File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int n: nums) {
            set.add(n);
       }
        
        int N = nums.length / 2;
        
        return Math.min(N, set.size());
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
# N/2 가져가도 좋다
def solution(nums):
    
    N = len(nums)
    map = dict.fromkeys(nums, 0)
    for n in nums:
        map[n] += 1
    
    if len(map.keys()) > N // 2:
        return N // 2
    
    return len(map.keys())